### ISSUE TITLE
PostCSS 8 

### ISSUE THREAD

**Issue by josemarluedke** [salsify/ember-css-modules#218]
Are there any plans for upgrading to PostCSS 8?

**Comment by Techn1x** [comment]
This is holding us back from postcss 8. All of our plugins are good to go. Since postcss 8 works with postcss7 plugins, Is it a big job to update or just a matter of bumping some dependencies?

**Comment by fpauser** [comment]
ember-css-modules depends on broccoli-css-modules which must be updated to postcss8. I did a pull-request which upgrades to postcss 8 https://github.com/salsify/broccoli-css-modules/pull/9/ which was never processed due to limmited resources and concerns about bumping ember-css-modules to V2.

**Comment by fpauser** [comment]
The solution in my current project was to use `ember-cli-postcss` together with `ember-css-modules`. With this setup you can use the latest and greatest postcss plugins by configuring `ember-cli-postcss` and run `ember-css-modules` without additional postcss plugins.

Note: As this project depended on `ember-cli-sass` I had to configure `ember-cli-postcss` accordingly, see https://github.com/jeffjewiss/ember-cli-postcss#switching-from-sass.

Build config `ember-cli-build.js`:
```js
module.exports = function(defaults) {
  var app = new EmberApp(defaults, {
    cssModules: {
      intermediateOutputPath: 'app/styles/_modules.css',
      plugins: {
        before: [postcssNested],
      },
    },
    postcssOptions: {
      compile: {
        extension: 'scss',
        enabled: true,
        parser: require('postcss-scss'),
        plugins: [
          {
            module: require('@csstools/postcss-sass'),
            options: {
              includePaths: [
                'node_modules'
              ]
            }
          },
          // keep for later, once we remove postcss-sass
          // {
          //   module: require('postcss-import'),
          //   options: {
          //     path: ['node_modules']
          //   }
          // },
          require('tailwindcss')('./config/tailwind.config.js')
        ]
      },
    }
  });
```

And then in `styles/app.scss`
```scss
@import 'modules';
```

Hope it helps.

**Comment by st-h** [comment]
@fpauser Thanks for sharing that. This is the config I have been using with a project that only uses plain css:

```
cssModules: {
  intermediateOutputPath: 'app/styles/app.css',
  plugins: {
    before: [postcssNested],
  },
},
postcssOptions: {
  compile: {
    enabled: true,
    plugins: [
      {
        module: require('postcss-import'),
        options: {
          path: ['node_modules']
        }
      },
    ]
  },
},
```

I also had to force PostCSS to version 8 in package.json, otherwise plugins would complain, even though I added them to the filter stage:

```
"resolutions": {
    "postcss": "8.2.10"
  },
```


Given all that, I think it would be great if we could be looking into resolving the issues of not being able to directly use PostCSS v8. 

How about we fork broccoli-css-modules and upgrade it to v8, if that is the current limitation? 

I am happy to help as much as I can. I'd just like to avoid having to depend on a fork of `ember-css-modules` in my projects.

**Comment by Alonski** [comment]
Would love to see this pushed forward as well :)

**Comment by gossi** [comment]
same here =)

**Comment by st-h** [comment]
@dfreeman what's your opinion on this? Do you see any way we can make this happen? Would love to remove the workarounds from my app again.

**Comment by dfreeman** [comment]
I just released [v1.4.0](https://github.com/salsify/ember-css-modules/releases/tag/v1.4.0) which should make things 'just work' with PostCSS 8 while remaining backwards compatible with older versions.

**Comment by st-h** [comment]
@dfreeman thanks so much 🙏 Works like a charm 🥳
